cmake/modules/CTags.cmake

```diff
@@ -0,0 +1,26 @@
+# after https://www.topbug.net/blog/2012/03/17/generate-ctags-files-for-c-slash-c-plus-plus-source-files-and-all-of-their-included-header-files/ and
+# https://stackoverflow.com/questions/9827208/run-a-shell-command-ctags-in-cmake-and-make
+
+set_source_files_properties(tags PROPERTIES GENERATED true)
+add_custom_target(ctags
```

Comment:
Shrug.  It can certainly be improved, but, having the 'ctags' command is a common requirement for being able to make tags, and this isn't a critical target for the build, it's volitional.  As for calculating the submodules...sure, that looks great, certainly better than hardcoded.  You wanna take this over, then?